wsclient/src/test/java/org/hpccsystems/ws/client/WSResroucesClientTest.java

```diff
@@ -88,7 +95,72 @@ public void serviceQueryTest()
     public void getContainerizedModeTest() throws Exception
     {
         System.out.println("Fetching isTargetHPCCContainerized...");
-        assertNotNull(client.isTargetHPCCContainerized());
+        assertNotNull(wsResourcesClient.isTargetHPCCContainerized());
+    }
+
+    @Test
+    public void testTargetQueryNullType() throws Exception
+    {
+        // Create a target query request
+        TargetQueryRequestWrapper request = new TargetQueryRequestWrapper();
+        // Set the request parameters
+        request.setType(null);
+
+        try
+        {
+            TargetQueryResponseWrapper response = wsResourcesClient.targetQuery(request);
+            Assert.assertNotNull(response);
+        }
+        catch (ArrayOfEspExceptionWrapper e)
+        {
+            List<EspExceptionWrapper> espexceptions = e.getEspExceptions();
+            Assert.assertNotNull(espexceptions);
+            Assert.assertNotNull(espexceptions.get(0));
+            Assert.assertTrue(e.getEspExceptions().get(0).getMessage().contains("HPCCQueueType not defined."));
```

Comment:
We could spell it out, but that's the whole point of these tests, it articulates the expected behavior to a given input.
The most valuable input from you is confirmation that this is the expected behavior of the ESP service method. If no queue type, it will report a failure as a soap exception.
Also, can you think of any other behavior we should be testing. The point is to test as much of the service as possible.